auth/standard_role_manager.cc

```diff
@@ -134,7 +134,20 @@ const resource_set& standard_role_manager::protected_resources() const {
 }
 
 bool standard_role_manager::safe(const cql3::statements::schema_altering_statement& stmt) const {
-    return !protected_resources().contains(make_data_resource(stmt.keyspace(), stmt.column_family()));
+    if (!protected_resources().contains(make_data_resource(stmt.keyspace(), stmt.column_family()))) {
+        return true;
+    }
+    if (auto alter = dynamic_cast<const cql3::statements::alter_table_statement*>(&stmt)) {
```

Comment:
this dynamic_cast is certainly reaching in too deeply into the smelly guts of cql3.